superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx

```diff
@@ -0,0 +1,1317 @@
+  if (!!recipients && recipientValue !== recipients) {
+    setRecipientValue(recipients);
+  }
+
+  const methodOptions = (options || []).map((method: NotificationMethod) => {
+    return (
+      <Select.Option key={method} value={method}>
+        {t(method)}
+      </Select.Option>
+    );
+  });
+
+  return (
+    <StyledNotificationMethod>
+      <div className="inline-container">
+        <StyledInputContainer>
+          <div className="input-container">
+            <Select
+              onChange={onMethodChange}
+              placeholder="Select Delivery Method"
+              defaultValue={method}
+              value={method}
+            >
+              {methodOptions}
+            </Select>
+          </div>
+        </StyledInputContainer>
+        {method !== undefined && !!onRemove ? (
+          <span
+            role="button"
+            tabIndex={0}
+            className="delete-button"
+            onClick={() => onRemove(index)}
+          >
+            <Icon name="trash" />
+          </span>
+        ) : null}
+      </div>
+      {method !== undefined ? (
+        <StyledInputContainer>
+          <div className="control-label">{t(method)}</div>
+          <div className="input-container">
+            <textarea
+              name="recipients"
+              value={recipientValue}
+              onChange={onRecipientsChange}
+            />
+          </div>
+        </StyledInputContainer>
+      ) : null}
+    </StyledNotificationMethod>
+  );
+};
+
+const AlertReportModal: FunctionComponent<AlertReportModalProps> = ({
+  addDangerToast,
+  onAdd,
+  onHide,
+  show,
+  alert = null,
+  isReport = false,
+}) => {
+  const [disableSave, setDisableSave] = useState<boolean>(true);
+  const [currentAlert, setCurrentAlert] = useState<AlertObject | null>();
+  const [isHidden, setIsHidden] = useState<boolean>(true);
+  const [contentType, setContentType] = useState<string>(
+    alert && alert.chart ? 'chart' : 'dashboard',
+  ); // Maybe make this always default to dashboard and don't update until alert is fetched
+  const [scheduleFormat, setScheduleFormat] = useState<string>(
+    'dropdown-format',
+  );
+
+  // Dropdown options
+  const [sourceOptions, setSourceOptions] = useState<MetaObject[]>([]);
+  const [dashboardOptions, setDashboardOptions] = useState<MetaObject[]>([]);
+  const [chartOptions, setChartOptions] = useState<MetaObject[]>([]);
+
+  const isEditMode = alert !== null;
+
+  // TODO: need to set status/settings list based on alert's notification settings
+  const [notificationAddState, setNotificationAddState] = useState<
+    NotificationAddStatus
+  >('active');
+  const [notificationSettings, setNotificationSettings] = useState<
+    NotificationSetting[]
+  >([]);
+
+  const onNotificationAdd = () => {
+    const settings: NotificationSetting[] = notificationSettings.slice();
+
+    settings.push({
+      recipients: '',
+      options: NOTIFICATION_METHODS, // Need better logic for this
```

Comment:
Do we still need this comment?

Reply:
Kinda. I wanted to update the options list based on whether the user already has email or slack selected, but I think that can be added later.

Reply:
ok, let's swap out that ambiguous comment for a `TODO:` ☝️